test/test_nans.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <gtest/gtest.h>
+#include <numeric>
+#include <type_traits>
+
+namespace {
+template <typename T> struct MeanTest : public ::testing::Test {
```

Comment:
`test/` contains generic test helper, I don't think we should add something so specific here?

Reply:
I had reservations, but it's used in two test suites (dataset and variable) and I wanted to avoid copying.